fix(app-layout): prevent display of empty app navigation panel

lorumic opened the pull request:
## Done

- Prevented display of an empty app navigation panel when passing no `navItems` and `undefined` as `sideNavigation` props.

## QA

### Storybook

To see rendered examples of all react-components, run:

```shell
yarn start
```

### QA in your project

from `react-components` run:

```shell
yarn build
npm pack
```

Install the resulting tarball in your project with:

```shell
yarn add <path-to-tarball>
```

### QA steps

- Check the new "No Navigation" story for ApplicationLayout in Storybook: it displays no app navigation at all.

### Percy steps

- There should be a new story in Storybook for ApplicationLayout, called "No Navigation". This should be the only visual difference.

webteam-app commented:
### [<img src='https://assets.ubuntu.com/v1/6baef514-ubuntu-circle-of-friends-large.svg' height=32 width=32> Demo</img>](https://react-components-1106.demos.haus)
### [<img src='https://assets.ubuntu.com/v1/e512b0e2-jenkins.svg' height=32 width=32> Jenkins </img>](https://jenkins.canonical.com/webteam/job/start-demo/5631)
### [<img src='https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg' height=32 width=32> demos.haus </img>](https://demos.haus)

huwshimi commented on src/components/ApplicationLayout/ApplicationLayout.tsx:
```diff
@@ -155,71 +155,74 @@ const ApplicationLayout = <
           }}
         />
       </AppNavigationBar>
```
The `AppNavigationBar` should also be hidden if there is no menu to display otherwise the top bar appears on mobile:

<img width="389" alt="Screenshot 2024-07-18 at 10 05 12 AM" src="https://github.com/user-attachments/assets/d7ef6de3-94dd-406b-a325-e316b3a31251">

huwshimi reviewed (commented):
I think some types should be updated as well. In BaseProps the logo type is set to NonNullable: `logo: NonNullable<PanelProps<PL>["logo"]>;` which can now be `logo?: PanelProps<PL>["logo"];`

Also because neither of the menu item props are require they should probably be made optional in:

```
export type Props<
  NI = SideNavigationLinkDefaultElement,
  PL = PanelLogoDefaultElement,
> = BaseProps<NI, PL> &
  ExclusiveProps<
    {
      navItems: NonNullable<SideNavigationProps<NI>["items"]>;
    },
    {
      sideNavigation: ReactNode;
    }
  >;
```

Probably adding `...?:` to both would be enough.

lorumic commented:
Thanks for the review @huwshimi 
I have addressed the points [here](https://github.com/canonical/react-components/compare/f4d4af60c2c01e62d4210c567b6b10aab2fa6577..2dde0a2d584272f7585eba84328ab27d5cb5683f)

huwshimi reviewed (approved):
Looks good, thanks for those fixes.
